File: remove-element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        k = 0
        for num in nums:
            if val != num:
                nums[k] = num
                k += 1
        return k


solution = Solution()
nums = [3, 2, 2, 3]
val = 3
print(solution.removeElement(nums, 3))
print(nums)

File: remove-element.c
#include <stdio.h>

int removeElement(int *nums, int numsSize, int val)
{
    int k = 0;
    for (int i = 0; i < numsSize; i++)
    {
        if (nums[i] != val)
        {
            nums[k] = nums[i];
            k++;
        }
    }
    return k;
}

int main()
{
    int nums[] = {3, 2, 2, 3};
    int numsSize = sizeof(nums) / sizeof(int);
    int val = 3;
    printf("\n%d\n", removeElement(nums, numsSize, val));
    for (int i = 0; i < numsSize; i++)
    {
        printf("%d, ", nums[i]);
    }

    return 0;
}